File: pset6/readability/readability.py
from cs50 import get_string

s = get_string("Text: ")

s_lower= s.lower()

letters_count= 0

words_count=0

sentences_count=0

for i in s_lower:
    
    
    if ord(i)>=97 and ord(i) <= 122:
        
        letters_count+=1
        
        
words_count = len(s_lower.split(' '))

    
if "mr." in s_lower:
        
    s_lower=s_lower.replace("mr.", '')
            
if "miss." in s_lower:
        
    s_lower= s_1.replace("miss.", '')
            
            
if "ms." in s_lower:
        
    s_lower= s_lower.replace("ms.", '')
            
if "mrs." in s_lower:
        
    s_lower= s_lower.replace("mrs.", '')
    
    
    
            
            
            
for i in s_lower:
    
    if i == '.' or i=='!' or i=='?':
        
        sentences_count+=1
        


Grade= round(0.0588*letters_count/(words_count/100) - 0.296*sentences_count/(words_count/100) -15.8 )


if Grade>=16:
    
    print("Grade 16+")
    
elif Grade<=1:
    
    print("Before Grade 1")
    
else:
    
    print(f"Grade {Grade}")
    







    
        
    



    


    






        
        

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>


int main(void) 
{

    string input_text= get_string("Text: ");
    int words= 0 , letters = 0, sentences =0 ;
    int n = strlen(input_text);
    int i, j;
    
    for (i = 0; i <= n ; i++)
    {
        
        
    
       if (n!=0)
       {
           if (input_text[i]== ' ' || input_text[i]== '\0')
           {
               words++;
           }
       } 
       else 
       {
           words=0;
       }
    
       if (input_text[i]>= 'A' && input_text[i]<= 'Z')
       {
        
        letters++;
       } 
       else if (input_text[i] >= 'a' && input_text[i] <= 'z')
       {
           letters++;
       }
       if (input_text[i]=='.' || input_text[i]=='!' || input_text[i]=='?')
       {
    
       if (input_text[i]=='.' && input_text[i-1]== 'r' && input_text[i-2]=='M'){
       break;
       } else if (input_text[i]=='.' && input_text[i-1]== 's' && input_text[i-2]=='r' && input_text[i-3]=='M'){
       break;
       } else if (input_text[i]=='.' && input_text[i-1]== 's' && input_text[i-2]=='M'){
       break;
       } else if (input_text[i]=='.' && input_text[i-1]== 's' && input_text[i-2]=='s' && input_text[i-3]=='i' && input_text[i-4]=='M'){
       break;
       }
    
    sentences++;
    
    }
    
    
    }


    
    
    
    float L = (100/ (float)words)*(letters), S= (100/(float)words)*(sentences);
    
    
    float index= 0.0588*L-0.296*S-15.8;
    
    int index_1 = round(index);
    
    
    
    
    if (index_1<1){
    printf("Before Grade 1\n");
    } else if (index_1>=16){
    printf("Grade 16+\n");
    } else {
    printf("Grade %i\n", index_1);
    }
    
    
    
    
}
